### PR TITLE
Add macurl2path stub

### PR DESCRIPTION
**Pull request by CraftSpider**
Added a version check for < 3.7, because that's where it exists on windows. Not sure if that's true on mac, or if this is really only 'supposed' to be on mac and it should instead be an OS check

### CODE REVIEW

**Comment by CraftSpider** [comment]
@hauntsaninja For the stubtest failures, do you think the module should be added to the whitelists for py37 and 38?

File: stdlib/3/macurl2path.pyi [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+import sys
+
+from typing import AnyStr, Union
+
+# Is this correct? It seems to be for windows
+if sys.version_info < (3, 7):
+    def url2pathname(pathname: AnyStr) -> AnyStr: ...
```
Comment (JelleZijlstra) [inline stdlib/3/macurl2path.pyi]: I think it just accepts str.

File: stdlib/3/macurl2path.pyi [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+import sys
+
+from typing import AnyStr, Union
+
+# Is this correct? It seems to be for windows
```
Comment (JelleZijlstra) [inline stdlib/3/macurl2path.pyi]: Looks like it exists on all platforms, but is meant to be used on Mac.

File: stdlib/3/macurl2path.pyi [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+if sys.version_info < (3, 7):
+    def url2pathname(pathname: AnyStr) -> AnyStr: ...
+    def pathname2url(pathname: AnyStr) -> AnyStr: ...
```
Comment (JelleZijlstra) [inline stdlib/3/macurl2path.pyi]: This one is also str -> str.

File: stdlib/3/macurl2path.pyi [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    def _pncomp2url(component: Union[str, bytes]) -> str: ...  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/3/macurl2path.pyi]: The rest of the module is undocumented too, to be fair.

**Comment by hauntsaninja** [comment]
Looks like it's for MacOS 9, and was removed completely in 3.7.
```
Misc/NEWS.d/3.7.0a1.rst
4636:Remove vestigial MacOS 9 macurl2path module and its tests.
```

Yes, it makes sense to add to the stubtest whitelists. #16 is the relevant issue for finding a right way to do it.

File: stdlib/3/macurl2path.pyi (hunk 1 above)
Comment (CraftSpider) [reply stdlib/3/macurl2path.pyi]: Yeah, it does. I didn't check at first and just followed urllib's types for these functions, but that was a mistake

**Pull request merged**